core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/groupBy.kt

```diff
@@ -21,14 +36,289 @@ import kotlin.reflect.KProperty
 // region DataFrame
 
 /**
+ * Groups the rows of this [DataFrame] based on the values in one or more specified [key columns][cols].
+ * Each unique value in a key column — or a unique tuple of values for multiple columns —
+ * defines a group consisting of all rows where the column(s) contain that value combination.
  *
- * @param cols key columns; Column for grouping can be created inplace
+ * Returns a [GroupBy] — a dataframe-like structure that contains all unique combinations of key values
+ * along with the corresponding groups of rows (each represented as a [DataFrame]).
  *
- * `df.groupBy { expr("columnName") { "someColumn"<Int>() + 15 } }`
+ * A [GroupBy] can then be:
+ * * [transformed][Transformation] into a new [GroupBy];
+ * * [reduced][Reducing] into a [DataFrame], where each group is collapsed into a single representative row;
+ * * [aggregated][Aggregation] into a [DataFrame], where each group is transformed into one or more rows of derived values;
+ * * [pivoted][Pivoting] into a [PivotGroupBy] structure, which combines [pivot] and [groupBy] operations.
  *
- * is equivalent to
+ * Grouping keys can also be created inline:
+ * ```kotlin
+ * // Create a new column "newName" based on existing "oldName" values
+ * // and use it as a grouping key:
+ * df.groupBy { expr("newName") { oldName.drop(5) } }
+ * ```
  *
- * `df.add("columnName") { "someColumn"<Int>() + 15 }.groupBy("columnName")`
+ * Check out [Grammar].
+ *
+ * @include [SelectingColumns.ColumnGroupsAndNestedColumnsMention]
+ *
+ * See [Selecting Columns][GroupBySelectingOptions].
+ *
+ * For more information: {@include [DocumentationUrls.GroupBy]}
+ *
+ * Don't confuse this with [group], which groups column into
+ * [column group][org.jetbrains.kotlinx.dataframe.columns.ColumnGroup].
+ */
+internal interface GroupByDocs {
+    /**
+     * ## [groupBy][groupBy] Operation Grammar
+     * {@include [LineBreak]}
+     * {@include [DslGrammarLink]}
+     * {@include [LineBreak]}
+     *
+     * ### Create and transform [GroupBy]
+     *
+     * [**`groupBy`**][groupBy]**`(`**`moveToTop: `[`Boolean`][Boolean]**` = true)  {  `**`columns: `[`ColumnsSelector`][ColumnsSelector]**` }`**
```

Comment:
`` ` = true)  {  ` `` requires an additional space in front of `=`. The https://github.github.com/gfm/#code-spans rules of MD state that code spans starting and ending with a space character will have one surrounding space trimmed, making the specific example look like `"moveToTop: Boolean= true) { columns:..."`. This is currently implemented differently in the KDoc markdown renderer of IntelliJ, but I suspect it's only a matter of time since they claim to follow the GFM standard.